src/detect-destination.c

```diff
@@ -0,0 +1,401 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "assert.h"
+
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-destination.h"
+
+static int DetectDestinationMatch(
+        DetectEngineThreadCtx *, Packet *, const Signature *, const SigMatchCtx *);
+static int DetectDestinationSetup(DetectEngineCtx *, Signature *, const char *);
+static void DetectDestinationFree(DetectEngineCtx *, void *);
+
+#ifdef UNITTESTS
+static void DetectDestinationRegisterTests(void);
+#endif
+
+/**
+ * \brief Registration function for destination: trusted/untrusted
+ *
+ * Only supporting roles of trusted/untrusted for faster matching.
+ * The destination being trusted/untrusted refers to the packet of
+ * a flow and the role of the live device that the packet is being
+ * copied to.
+ */
+void DetectDestinationRegister(void)
+{
+    sigmatch_table[DETECT_DESTINATION].name = "destination";
+    sigmatch_table[DETECT_DESTINATION].desc =
+            "Match the destination of the traffic based on the role "
+            "(trusted/untrusted) of the interface.";
+    sigmatch_table[DETECT_DESTINATION].Match = DetectDestinationMatch;
```

Comment:
once we have the keyword docs, then we can add its URL reference here with `url` attribute